platform/darwin/src/MGLFeature.h

```diff
@@ -18,15 +18,19 @@ NS_ASSUME_NONNULL_BEGIN
  You can add custom data to display on the map by creating feature objects and
  adding them to an `MGLShapeSource` using the
  `-[MGLShapeSource initWithIdentifier:shape:options:]` method or
- `MGLShapeSource.shape` property. Similarly, you can add `MGLPointFeature`,
- `MGLPolylineFeature`, and `MGLPolygonFeature` objects to the map as annotations
- using `-[MGLMapView addAnnotations:]` and related methods.
+ `MGLShapeSource.shape` property.
 
  In addition to adding data to the map, you can also extract data from the map:
  `-[MGLMapView visibleFeaturesAtPoint:]` and related methods return feature
  objects that correspond to features in the source. This enables you to inspect
  the properties of features in vector tiles loaded by `MGLVectorSource` objects.
  You also reuse these feature objects as overlay annotations.
+
+ While it is possible to add `MGLFeature`-conforming objects to the map as
+ annotations using `-[MGLMapView addAnnotations:]` and related methods, doing so
+ will convert feature objects into plain annotations. Features added as
+ annotations will not have `identifier` or `attributes` properties set and
+ should not be used be with feature querying.
```

Comment:
Gave this a go in 5fabed1:

<img width="870" alt="screen shot 2017-07-19 at 3 21 57 pm" src="https://user-images.githubusercontent.com/1198851/28385426-0c044e56-6c96-11e7-8511-8c2b95019b80.png">